### PR TITLE
[SYCL] Process less bindless images code in handler/queue headers

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Keep the amount of code processed by the C++ front end at absolute minimum. Part of ongoing efforts to reduce the amount of the SYCL code required to process when including `sycl/detail/core.hpp`.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
ping everybody.

**Review by przemektmalon — APPROVED** [review]
The bindless images tests are failing with this branch as it doesn't incorporate the recently merged fix in https://github.com/intel/llvm/pull/13585

They work with that fix, though, so if this doesn't undo anything from the PR above, it LGTM.

File: sycl/include/sycl/handler.hpp [hunk 1]
```diff
@@ -161,6 +160,12 @@ template <class _name, class _dataT, int32_t _min_capacity, class _propertiesT,
 class pipe;
 }
 
+namespace ext ::oneapi ::experimental {
+// State of a queue with regards to graph recording,
+// returned by info::queue::state
+struct image_descriptor;
```
Comment (Seanst98) [inline sycl/include/sycl/handler.hpp:166]: This comment seems incorrectly applied to the `image_descriptor`. Did it not originally belong to the `queue_state` declaration?

```
// State of a queue with regards to graph recording,
// returned by info::queue::state
enum class queue_state { executing, recording };
```

**Review by Seanst98 — CHANGES_REQUESTED** [review]

**Review by steffenlarsen — APPROVED** [review]
Aside from the `image_descriptor` comment, I think this looks good.

File: sycl/include/sycl/handler.hpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/include/sycl/handler.hpp]: That's right, thank you.

File: sycl/include/sycl/ext/oneapi/bindless_images_memory.hpp [hunk 2]
```diff
@@ -12,8 +12,9 @@
 #include <sycl/detail/export.hpp>                         // for __SYCL_EXPORT
 #include <sycl/device.hpp>                                // for device
 #include <sycl/ext/oneapi/bindless_images_descriptor.hpp> // for image_desc...
-#include <sycl/image.hpp>                                 // for image_chan...
-#include <sycl/range.hpp>                                 // for range
+#include <sycl/ext/oneapi/bindless_images_mem_handle.hpp>
```
[the remark this answers is not in the document]
Comment (isaacault) [reply sycl/include/sycl/ext/oneapi/bindless_images_memory.hpp:15]: Sounds good to me.

**Review by isaacault — APPROVED** [review]

**Pull request merged**